projects/plugins/crm/src/automation/class-automation-workflow.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+namespace Automattic\Jetpack\CRM\Automation;
+
+class Automation_Workflow {
+	
+	/** @var int */
+	private $id;
+	
+	/** @var string */
+	public $name;
+	
+	/** @var string */
+	public $description;
+	
+	/** @var string */
+	public $category;
+	
+	/** @var array */
+	public $triggers;
+	
+	/** @var array */
+	public $initial_step;
+	
+	/** @var bool */
+	public $active;
+	
+	/** @var Automation_Engine */
+	private $automation_engine;
+	
+	/** @var Automation_Logger */
+	private $logger;
+	
+	public function __construct( array $workflow_data, Automation_Engine $automation_engine ) {
+		$this->id           = $workflow_data['id'] ?? null;
+		$this->triggers     = $workflow_data['triggers'] ?? array();
+		$this->initial_step = $workflow_data['initial_step'] ?? array();
+		$this->name         = $workflow_data['name'];
+		$this->description  = $workflow_data['description'] ?? '';
+		$this->category     = $workflow_data['category'] ?? '';
+		$this->active       = $workflow_data['is_active'] ?? true;
+		
+		$this->automation_engine = $automation_engine;
+		$this->logger            = $automation_engine->get_logger() ?? Automation_Logger::instance();
+	}
+	
+	/**
+	 * Get the id of this workflow
+	 * 
+	 * @return int
+	 */
+	public function get_id(): int {
+		return $this->id;
+	}
+
+	public function set_triggers( array $triggers ) {
+		$this->triggers = $triggers;
+	}
+
+	/**
+	 * Get the trigger names of this workflow
+	 * 
+	 * @return array
+	 */
+	public function get_triggers(): array {
+		return $this->triggers;
+	}
+
+	/**
+	 * Instance the triggers of this workflow
+	 * @throws Workflow_Exception
+	 */
+	public function init_triggers() {
+		
+		if ( ! $this->is_active() ) {
+			return;
+		}
+		
+		foreach ( $this->get_triggers() as $trigger_name ) {
+			try {
+				$class_name = $this->automation_engine->get_trigger_class( $trigger_name );
+				
+				/** @var Trigger $trigger */
+				$trigger = new $class_name();
+				$trigger->init( $this );
+				
+			} catch ( Automation_Exception $e ) {
+				throw new Workflow_Exception( sprintf( __( 'An error happened initializing the trigger. %s', 'zero-bs-crm' ),  $e->getMessage() ) );
+			}
+		}
+	}
+	
+	/** 
+	 * Set initial step of this workflow
+	 * @param array $step_data
+	 */
+	public function set_initial_step( array $step_data ) {
+		$this->initial_step = $step_data;
+	}
+
+	/**
+	 * Get the workflow as an array
+	 * @return array
+	 */
+	public function get_workflow_array() {
+		
+		$workflow = array(
+			'name'          => $this->name,
+			'description'   => $this->description,
+			'category'      => $this->category,
+			'is_active'     => $this->active,
+			'triggers'      => $this->triggers,
+			'initial_step'  => $this->initial_step,
+		);
+
+		return $workflow;
+	}
+
+	/**
+	 * Get the initial step of this workflow
+	 * 
+	 * @return array
+	 */
+	public function get_initial_step(): array {
```

Comment:
This method doesn't seem to be used anywhere 😄

Reply:
For now... still in progress.